spark/dl/src/main/resources/serialization/bigdl.proto

```diff
@@ -54,9 +57,27 @@ message InitMethod {
 message BigDLTensor{
     DataType datatype = 1;
     repeated int32 size = 2; //size of tensor
-    repeated float float_data = 3; //tensor element
-    repeated double double_data = 4; //tensor element
+    repeated int32 stride = 3; //stride of tensor
+    int32 offset = 4; //offset
+    int32 dimension = 5; // dimension
+    int32 nElements = 6; //total number of elements
+    bool isScalar = 7; // is scalar tensor
+    TensorStorage storage = 8; // reference to storage
+    int32 id = 9; // tensor ID, used for tensor sharing
 }
+
+message TensorStorage {
+    DataType datatype = 1;
+    repeated float float_data = 2; //float tensor element
+    repeated double double_data = 3; //double tensor element
+    repeated bool bool_data = 4; //boolean tensor element
+    repeated string string_data = 5; //string tensor element
+    repeated int32 int_data = 6; //int tensor element
+    repeated int64 long_data = 7; //long tensor element
+    repeated bytes bytes_data = 8; //byte tensor element
```

Comment:
Do we need `repeated` for `bytes` here?

Reply:
bytes is mapped to ByteString which is the type of tensor, so a ByteString typed tensor has multiple 'bytestring'